src/fileio.cpp

```diff
@@ -1111,13 +1119,22 @@ void DeterminePaths(const char *exe)
 {
 	DetermineBasePaths(exe);
 
-#if defined(WITH_XDG_BASEDIR) && defined(WITH_PERSONAL_DIR)
-	const char *xdg_config_home = xdgConfigHome(nullptr);
-	std::string config_home(xdg_config_home);
-	config_home += PATHSEP;
-	config_home += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
-	free(xdg_config_home);
-
+#ifdef USE_XDG
+	std::string config_home;
+	const std::string homedir = GetHomeDir();
+	const char *xdg_config_home = getenv("XDG_CONFIG_HOME");
+	if (xdg_config_home != nullptr) {
+		config_home = xdg_config_home;
+		config_home += PATHSEP;
+		config_home += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
+	} else if (!homedir.empty()) {
+		/* Defaults to ~/.config */
+		config_home = homedir;
+		config_home += PATHSEP;
+		config_home += ".config";
+		config_home += PATHSEP;
```

Comment:
nearly the same code as `XDG_DATA_HOME` but written in a different way. Feels a bit silly.